packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1871,6 +1920,20 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
       this.validateSamlAuthenticationOptions(props.fineGrainedAccessControl?.samlAuthenticationOptions);
     }
 
+    if (props.capacity?.nodeOptions) {
+      // Validate coordinator node configuration
+      const coordinatorConfig = props.capacity.nodeOptions.find(opt => opt.nodeType === 'coordinator')?.nodeConfig;
```

Comment:
Does this mean that `nodeOptions` can only contain one element with `nodeType: 'coordinator'`?
If so, we should validate that as well.

Reply:
Currently yes, and each config can be different, changed this to - 

`const coordinatorConfig = props.capacity.nodeOptions.find(opt => opt.nodeType === NodeType.COORDINATOR)?.nodeConfig;`